js/coinfloor.js

```diff
@@ -201,6 +201,139 @@ module.exports = class coinfloor extends Exchange {
+            }
+        }
+        const request = {
+            'id': market['id'],
+            'limit': limit,
+        };
+        const response = await this.privatePostIdUserTransactions (this.extend (request, params));
+        return this.parseLedger (response, undefined, since, limit);
+    }
+
+    parseLedgerEntryStatus (status) {
+        const types = {
+            'completed': 'ok',
+        };
+        return this.safeString (types, status, status);
+    }
+
+    parseLedgerEntryType (type) {
+        const types = {
+            '0': 'transaction', // deposit
+            '1': 'transaction', // withdrawal
+            '2': 'trade',
+        };
+        return this.safeString (types, type, type);
+    }
+
+    parseLedgerEntry (item, currency = undefined) {
+        // trade:
+        //   {
+        //     "datetime": "2017-07-25 06:41:24",
+        //     "id": 1500964884381265,
+        //     "type": 2,
+        //     "xbt": "0.1000",
+        //     "xbt_eur": "2322.00",
+        //     "eur": "-232.20",
+        //     "fee": "0.00",
+        //     "order_id": 84696745 }
+        //
+        // transactions:
+        //   {
+        //     "datetime": "2017-07-25 13:19:46",
+        //     "id": 97669,
+        //     "type": 1,
+        //     "xbt": "-3.0000",
+        //     "xbt_eur": null,
+        //     "eur": "0",
+        //     "fee": "0.0000",
+        //     "order_id": null }
+        //   {
+        //     "datetime": "2017-07-27 16:44:55",
+        //     "id": 98277,
+        //     "type": 0,
+        //     "xbt": "0",
+        //     "xbt_eur": null,
+        //     "eur": "4970.04",
+        //     "fee": "0.00",
+        //     "order_id": null }
+        const keys = Object.keys (item);
+        let baseId = undefined;
+        let quoteId = undefined;
+        let baseAmount = undefined;
+        let quoteAmount = undefined;
+        for (let i = 0; i < keys.length; i++) {
+            const key = keys[i];
+            if (key.indexOf ('_') > 0) {
+                const parts = key.split ('_');
+                const numParts = parts.length;
+                if (numParts === 2) {
+                    const tmpBaseAmount = this.safeFloat (item, parts[0]);
+                    const tmpQuoteAmount = this.safeFloat (item, parts[1]);
+                    if (tmpBaseAmount !== undefined && tmpQuoteAmount !== undefined) {
+                        baseId = parts[0];
+                        quoteId = parts[1];
+                        baseAmount = tmpBaseAmount;
+                        quoteAmount = tmpQuoteAmount;
+                    }
+                }
+            }
+        }
+        if (this.safeString (item, 'type') === '2') {
+            // it's a trade so let make 2 entries
+            const orderId = this.safeString (item, 'order_id');
+            return [
+                this.createLedgerItem (baseAmount, baseId, orderId, undefined, item),
```

Comment:
Don't we lose fee info here?

Reply:
Another issue is that the `id` has to be unique...

Reply:
The id is unique.  The reference id joins together 2 sides of the order.

Reply:
> Don't we lose fee info here?

Er, yes.  My mistake.

Reply:
> The id is unique.

No, it was not unique. The ledger entry id in your code was duplicated for the two sides of the trade.

> The reference id joins together 2 sides of the order.

The `trade` ledger entry should have a trade id (not an order id) in the `referenceId` and a unique `id`. In your case, the `id` was duplicated.